mergify_engine/web/api/queues.py

```diff
@@ -317,5 +358,309 @@ async def repository_queues(
                 )
 
             queues.queues.append(queue)
-
         return queues
+
+
+@router.put(
+    "/repos/{owner}/{repository}/queue/{queue_name}/freeze",  # noqa: FS003
+    summary="Freezes merge queue",
+    description="Freezes the merge of the requested queue and the queues following it",
+    response_model=QueueFreezeResponse,
+)
+async def create_queue_freeze(
+    queue_freeze_payload: QueueFreezePayload,
+    application: application_mod.Application = fastapi.Depends(  # noqa: B008
+        security.get_application
+    ),
+    owner: github_types.GitHubLogin = fastapi.Path(  # noqa: B008
+        ..., description="The owner of the repository"
+    ),
+    repository: github_types.GitHubRepositoryName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the repository"
+    ),
+    queue_name: rules.QueueName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the queue"
+    ),
+    redis_cache: utils.RedisCache = fastapi.Depends(  # noqa: B008
+        redis.get_redis_cache
+    ),
+    redis_queue: utils.RedisQueue = fastapi.Depends(  # noqa: B008
+        redis.get_redis_queue
+    ),
+    installation_json: github_types.GitHubInstallation = fastapi.Depends(  # noqa: B008
+        security.get_installation
+    ),
+) -> QueueFreezeResponse:
+
+    if queue_freeze_payload.reason == "":
+        queue_freeze_payload.reason = "No freeze reason was specified."
+
+    async with github.aget_client(installation_json) as client:
+        try:
+            # Check this token as access to this repository
+            repo = typing.cast(
+                github_types.GitHubRepository,
+                await client.item(f"/repos/{owner}/{repository}"),
+            )
+        except (http.HTTPNotFound, http.HTTPForbidden, http.HTTPUnauthorized):
+            raise fastapi.HTTPException(status_code=404)
+
+        sub = await subscription.Subscription.get_subscription(
+            redis_cache, repo["owner"]["id"]
+        )
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+        repository_ctxt = installation.get_repository_from_github_data(repo)
+        config_file = await repository_ctxt.get_mergify_config_file()
+
+        if config_file is None:
+            raise fastapi.HTTPException(
+                status_code=404, detail="Mergify configuration file is missing."
+            )
```

Comment:
Do this before getting the `sub`; it'd avoid retrieving the sub if there's no config file.